glean-core/ios/Glean/Glean.swift

```diff
@@ -73,90 +73,84 @@ public class Glean {
             return
         }
 
-        self.registerPings(Pings.shared)
-
         self.configuration = configuration
+        setUploadEnabled(uploadEnabled)
+
+        // Execute startup off the main thread
+        Dispatchers.shared.launchConcurrent {
```

Comment:
I'm a bit confused by the different queues on iOS, but I guess that's due to my lack of knowledge of this platform. On the Kotlin side, the whole init runs in the `Dispatchers.API` context (as the first thing after the currently executed operation, if any) preventing any other `Dispatchers.API.launch` call from being executed. Is this the case here? Can things executed with `Dispatchers.shared.launchConcurrent` run at the same time of `Dispatchers.shared.serialOperationQueue`?

Reply:
The concurrent queue is just that, it allows more than one task to run on it concurrently.  I'm using that particular queue here because if we added this to the 'serial' queue, any serial tasks that needed to execute during the init would be blocked until the init returned, which would not allow us to run and await the flushing of the queued tasks (or any other `launchAPI` block)

Reply:
After re-reading your question, allow me to expand on this a bit:  during the init, and up until we flush the queue, any `launchAPI` tasks would be just added to the queue.  Since the flushing of the queue happens almost at the end of `initialize`, it could be possible for tasks that run via `launchAPI` on the serial queue while `initialize` is sending the "dirty_startup" baseline ping, registering the lifecycle observers, or sending the deletion request ping.  If you see any problem with this, please let me know as I will have to change how `flushQueuedInitialTasks` works because, as I mentioned in the previous comment, executing `initialize` on the serial queue would prevent executing any of the queued tasks until after `initialize` returned, and we wouldn't be able to await them.  In this case, we could just execute them synchronously, I suppose...

Reply:
Worth noting: The only other call to `launchConcurrent` left is inside the uploader to upload already written data in a thread.
The only way to get there is by sending a ping, which itself is gated by a `launchAPI` (for all external users at least). Thus there's no other `launchCorrent` happening before this task here flushes the queue.

Reply:
I therefore think using this queue for the init and the other queue for the normal tasks is the correct usage.

Reply:
@travis79 can you add a comment about what @badboy said?

Reply:
Yes, sorry but this is already merged.  I'll open a new PR and add some usage notes about the use of `launchConcurrent` vs `launchAPI`

Reply:
> @travis79 can you add a comment about what @badboy said?

[Bug filed to take care of the documentation](https://bugzilla.mozilla.org/show_bug.cgi?id=1615603)